src/Psalm/Type/UnionTrait.php

```diff
@@ -795,9 +797,20 @@ public function hasMixed(): bool
     /**
      * @psalm-mutation-free
      */
-    public function isMixed(): bool
+    public function isMixed(bool $check_templates = false): bool
     {
-        return isset($this->types['mixed']) && count($this->types) === 1;
+        return count(
+            array_filter(
+                $this->types,
+                static fn($type, $key): bool => $key == 'mixed'
```

Comment:
Can you use === please? I see it's not used everywhere, but IMHO it should be :p